lib/schedule.js

```diff
@@ -186,9 +186,31 @@ Job.prototype.schedule = function(spec) {
   var self = this;
   var success = false;
   var inv;
+  var start;
+  var end;
+
+  if (typeof spec === 'object' && spec.rule) {
+    start = spec.start || null;
+    end = spec.end || null;
+    spec = spec.rule;
+
+    if (start != null) {
+      if (!(start instanceof Date)) {
+        start = new Date(start);
+      }
+      if (!isValidDate(start) || start.getTime() < Date.now()) {
+        start = null;
```

Comment:
It can be a good idea adding the `(date instanceof Date)` check inside `isValidDate()` function as they're always used together. It would simplify the code a little.